### DISCUSSION TITLE
rdfiltercatalog boost warning question

### DISCUSSION THREAD

**Discussion by vfscalfani** [rdkit/rdkit#7587] [Q&A] [answered]
I have adapted some code from this tutorial to filter molecules by substructure: https://github.com/greglandrum/AIDD_RDKit_Tutorial_2021/blob/main/RDKit_Intro.ipynb

I'm using RDKit v2024.03.4, conda-forge, Linux x86. Here is what I am doing:

```python
from rdkit import Chem
from rdkit.Chem import FunctionalGroups
_ = FunctionalGroups.BuildFuncGroupHierarchy()

# names and SMARTS patterns of all functional groups
for key, value in FunctionalGroups.groupDefns.items():
    smartsPattern = value.smarts  # Access the SMARTS pattern
    # print(f"{key}: {smartsPattern}")

# Filter by Carboxylic Acid
from rdkit.Chem import rdfiltercatalog
pattern = FunctionalGroups.groupDefns['CarboxylicAcid'].smarts
matcher = rdfiltercatalog.SmartsMatcher('CarboxylicAcid_filter', pattern, 1, 1)   # 1,1, is min, max count

```

Warning message output:

```
<frozen importlib._bootstrap>:488: RuntimeWarning: to-Python converter for boost::shared_ptr<RDKit::FilterMatcherBase> already registered; second conversion method ignored.
```

The next step of using the setup matcher (e.g., `matcher.HasMatch(mol)`) works fine, but I am curious if there is something I can change to avoid this warning? 

Thanks,

Vin

**Comment by bp-kelley** [comment, marked as answer]
Issue referenced in #7585

**Reply by vfscalfani** [reply]
Thanks very much, I missed that. 

Vin

**Comment by bp-kelley** [comment]
I think they were added at the same time.

There is a pr in testing, thanks for the report.
